run_unittests.py

```diff
@@ -510,7 +510,8 @@ def tearDown(self):
                 windows_proof_rmtree(path)
             except FileNotFoundError:
                 pass
-        os.environ = self.orig_env
+        os.environ.clear()
+        os.environ.update(self.orig_env)
```

Comment:
I'm curious, why is this needed?

Reply:
`os.environ` is a dict like object but not actually a dict. `setUp` and `tearDown` previously tried to restore the state of `os.environ` by assigning `os.environ.copy()` to `os.environ'. But that is a normal dict and thus after that `os.environ` no longer has effect on processes started with subprocess.

I spend some time debugging why the test worked in isolation but not with the whole test suit and decided to fix the root cause, by not assigning but by replacing the contents back into the special object.